Utilities/AltitudeAngel/AltitudeAngel.cs

```diff
@@ -1,32 +1,46 @@
 ﻿using System;
-using System.Text;
-using System.Windows.Forms;
+using System.Threading.Tasks;
 using AltitudeAngelWings;
-using AltitudeAngelWings.ApiClient.Client;
+using AltitudeAngelWings.Extra;
 using AltitudeAngelWings.Service;
-using AltitudeAngelWings.Service.FlightData;
-using AltitudeAngelWings.Service.FlightData.Providers;
-using AltitudeAngelWings.Service.Messaging;
-using MissionPlanner.Controls;
+using MissionPlanner.GCSViews;
 
 namespace MissionPlanner.Utilities.AltitudeAngel
 {
-    public class AltitudeAngel : IDisposable
+    internal static class AltitudeAngel
     {
-        public static MissionPlannerAdaptor MP = new MissionPlannerAdaptor();
-        private static MessagesService Message = new MessagesService();
-
-        public static AltitudeAngelService service = null;
+        internal static void Configure()
+        {
+            AltitudeAngelPlugin.Configure();
+            ServiceLocator.Register<IMissionPlanner>(l => new MissionPlannerAdaptor(
+                new MapAdapter(FlightData.instance.gMapControl1),
+                new MapAdapter(FlightPlanner.instance.MainMap),
+                () => FlightPlanner.instance.GetFlightPlanLocations()));
+            ServiceLocator.Register<IMissionPlannerState>(l => new MissionPlannerStateAdapter(

```

Comment:
im pretty sure this reference will fail, as its using a class that will be replaced at multiple levels

Reply:
I didn't notice it was a dynamically referenced function, so I think its ok

Reply:
Yes, that's correct. MissionPlannerStateAdapter takes a Func<CurrentState> and this is called regularly (twice a second at the moment) to get the current state.